src/main/scala/mesosphere/marathon/api/AuthResource.scala

```diff
@@ -17,25 +16,51 @@ trait AuthResource extends RestResource {
   implicit val authenticator: Authenticator
   implicit val authorizer: Authorizer
 
-  def authenticated(request: HttpServletRequest)(fn: Identity => Response): Response = {
+  /**
+    * Authenticate an HTTP request, asynchronously.
+    *
+    * @return If succeed, future with identity. If failed, returns failed future with a RejectionException.
+    */
+  def authenticatedAsync(request: HttpServletRequest): Future[Identity] = {
```

Comment:
I'd actually changes this to 

```scala
def authenticatedAsync(request: HttpServletRequest, response: )(fn: Identity => Future[Unit]) = sendResponse(response) { async {
  identity = await(authenticationRequest.transform{ ... })
  await(fn(identity))
}
}
```

this will reduce the changes and simplify things.